src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java

```diff
@@ -0,0 +1,64 @@
+package org.wso2.carbon.connector.pcml;
+
+import com.ibm.as400.access.AS400ConnectionPool;
+
+public class PCMLPool implements PCMLPoolMBean {
+
+    AS400ConnectionPool as400ConnectionPool;
+    int maxConnections;
+    long maxInactivity;
+    int maxLifetime;
+    int maxUseCount;
+    int maxUseTime;
+    boolean isRunMaintenance;
+    boolean isThreadUsed;
+    int cleanupInterval;
+    boolean pretestConnections;
+
+
+    public PCMLPool(AS400ConnectionPool connectionPool) {
+        this.as400ConnectionPool = connectionPool;
+    }
+
+    @Override
+    public int getMaxConnections() {
+        return this.getAs400ConnectionPool().getMaxConnections();
+    }
+
+    public AS400ConnectionPool getAs400ConnectionPool() {
```

Comment:
Let's make this method private or use `this.as400ConnectionPool` in the rest of the methods